src/lib/expression/evaluation/expression_evaluator.cpp

```diff
@@ -71,15 +71,67 @@ std::shared_ptr<AbstractExpression> rewrite_between_expression(const AbstractExp
   // (This is desirable because three expression data types (from three arguments) generate many type
   // combinations and thus lengthen compile time and increase binary size notably.)
 
-  const auto* between_expression = dynamic_cast<const BetweenExpression*>(&expression);
-  Assert(between_expression, "Expected BetweenExpression");
+  const auto* between_inclusive_expression = dynamic_cast<const BaseBetweenExpression*>(&expression);
+  Assert(between_inclusive_expression, "Expected Between Expression");
+  Assert(between_inclusive_expression->predicate_condition == PredicateCondition::BetweenInclusive,
+         "Expected BetweenInclusive Predicate Condition");
 
-  const auto gte_expression = greater_than_equals_(between_expression->value(), between_expression->lower_bound());
-  const auto lte_expression = less_than_equals_(between_expression->value(), between_expression->upper_bound());
+  const auto gte_expression =
+      greater_than_equals_(between_inclusive_expression->value(), between_inclusive_expression->lower_bound());
+  const auto lte_expression =
+      less_than_equals_(between_inclusive_expression->value(), between_inclusive_expression->upper_bound());
 
   return and_(gte_expression, lte_expression);
 }
 
+std::shared_ptr<AbstractExpression> rewrite_between_lower_exclusive_expression(const AbstractExpression& expression) {
```

Comment:
Please use a single method for rewriting the Between, this is just too much code duplication right now

```c++

std::shared_ptr<AbstractExpression> rewrite_between_expression(const AbstractExpression& expression) {
  const auto* between_expression = dynamic_cast<const BetweenExpression*>(&expression);
  Assert(between_expression, "Expected Between Expression");

  const auto predicate_condition = between_expression->predicate_condition;
  
  const auto lower_bound_condition = is_between_lower_inclusive(predicate_condition) ? 
                                         PredicateCondition::GreaterThanEquals : PredicateCondition::GreaterThan;
  const auto upper_bound_condition = is_between_upper_inclusive(predicate_condition) ? 
                                         PredicateCondition::LessThanEquals : PredicateCondition::LessThan;

  const auto lower_bound = std::make_shared<BinaryPredicateExpression>(lower_bound_condition, 
    between_expression->value(), between_expression->lower_bound());
  const auto upper_bound = std::make_shared<BinaryPredicateExpression>(upper_bound_condition, 
    between_expression->value(), between_expression->lower_bound());

  return and_(lower_bound, upper_bound);
}
```